### PR TITLE
video: improve error logging in `IGPUCommandBuffer`

### PR DESCRIPTION
**Pull request by alichraghi**

### CODE REVIEW

File: include/nbl/video/IGPUCommandBuffer.h [hunk 1]
```diff
@@ -737,14 +737,20 @@ class NBL_API2 IGPUCommandBuffer : public IBackendObject
             if (invalidBufferBinding({range.offset,range.buffer},alignment,usages))
                 return true;
             if ((range.size&(alignment-1)) && range.size!=asset::SBufferRange<IGPUBuffer>::WholeBuffer)
+            {
+                m_logger.log("Size %d not aligned to %d for the command!", system::ILogger::ELL_ERROR, range.size, alignment);
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/IGPUCommandBuffer.h:741]: you want to use a macro like in `ILogicalDevice` so that `__LINE__` `__PRETTY_FUNCTION__` are printed

Comment (devshgraphicsprogramming) [reply include/nbl/video/IGPUCommandBuffer.h:741]: you can undef before the header ends to stop pollution

File: src/nbl/video/IGPUCommandBuffer.cpp [hunk 2]
```diff
@@ -1256,7 +1584,10 @@ bool IGPUCommandBuffer::drawIndirectCount(const asset::SBufferBinding<const IGPU
         return false;
 
     if (!m_cmdpool->m_commandListPool.emplace<IGPUCommandPool::CDrawIndirectCountCmd>(m_commandList, core::smart_refctd_ptr<const IGPUBuffer>(indirectBinding.buffer), core::smart_refctd_ptr<const IGPUBuffer>(countBinding.buffer)))
-        return false;
+    {
+        NBL_LOG_ERROR("out of host memory!");
+        return true;
+    }
```
Comment (devshgraphicsprogramming) [inline src/nbl/video/IGPUCommandBuffer.cpp:1590]: why are you now returning a different value?

Please do some eyeballing of all your changes and make sure you don't change the return values

File: include/nbl/video/IGPUCommandBuffer.h [hunk 3]
```diff
@@ -11,6 +11,10 @@
 #include "nbl/video/IGPUCommandPool.h"
 #include "nbl/video/IQueue.h"
 
+#include "git_info.h"
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/IGPUCommandBuffer.h:14]: you don't need to include `git_info`

Comment (alichraghi) [reply include/nbl/video/IGPUCommandBuffer.h:14]: it's required by `logging_macros.h`

File: src/nbl/video/IGPUCommandBuffer.cpp (hunk 2 above)
Comment (alichraghi) [reply src/nbl/video/IGPUCommandBuffer.cpp]: > Please do some eyeballing of all your changes and make sure you don't change the return values
done

File: include/nbl/video/IGPUCommandBuffer.h (hunk 3 above)
Comment (devshgraphicsprogramming) [reply include/nbl/video/IGPUCommandBuffer.h:14]: @AnastaZIuk is it allowed to be a public header and what include path must it have?

Comment (devshgraphicsprogramming) [reply include/nbl/video/IGPUCommandBuffer.h:14]: i kinda don't like we include it without a folder path or anything (think what could happen if another project had same named include)

Comment (devshgraphicsprogramming) [reply include/nbl/video/IGPUCommandBuffer.h:14]: @AnastaZIuk ?

File: src/nbl/video/IGPUCommandBuffer.cpp [hunk 4]
```diff
@@ -16,27 +19,27 @@ bool IGPUCommandBuffer::checkStateBeforeRecording(const core::bitflag<queue_flag
 {
     if (m_state!=STATE::RECORDING)
     {
-        m_logger.log("Failed to record into command buffer: not in RECORDING state.", system::ILogger::ELL_ERROR);
+        NBL_LOG_ERROR("not in RECORDING state!");
```
Comment (devshgraphicsprogramming) [inline src/nbl/video/IGPUCommandBuffer.cpp:22]: why remove the "failed to record into command buffer" ?

**Comment by devshgraphicsprogramming** [comment]
@alichraghi how many examples did you test this with, also both RWDI and Debug ?

**Comment by alichraghi** [comment]
don't remember how many but pretty sure only ran in Debug mode

**Comment by devshgraphicsprogramming** [comment]
> don't remember how many but pretty sure only ran in Debug mode

try a few in RWdi and Release

**Comment by alichraghi** [comment]
done. all good

**Pull request merged**